src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -1662,14 +1662,34 @@ public InputStream downloadStream(String fileName) throws SnowflakeSQLException
 
     remoteLocation remoteLocation = extractLocationAndPath(stageInfo.getLocation());
 
-    String stageFilePath = fileName;
+    // when downloading files as stream there should be only one file in source files
+    String sourceLocation =
+        sourceFiles.stream()
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new SnowflakeSQLException(
+                        queryID,
+                        SqlState.NO_DATA,
+                        ErrorCode.FILE_NOT_FOUND.getMessageCode(),
+                        session,
+                        "File not found: " + fileName));
+
+    String stageFilePath;
+    if (fileName.equals(sourceLocation)) {
+      stageFilePath = fileName;
+    } else {
+      // filename may be different from source location e.g. in git repositories
+      logger.debug("Changing file to download location from {} to {}", fileName, sourceLocation);
+      stageFilePath = sourceLocation;
+    }
 
     if (!remoteLocation.path.isEmpty()) {
-      stageFilePath = SnowflakeUtil.concatFilePathNames(remoteLocation.path, fileName, "/");
+      stageFilePath = SnowflakeUtil.concatFilePathNames(remoteLocation.path, sourceLocation, "/");
```

Comment:
Could we add log for the final value of stagePathValue?

Reply:
I adde more logs